### PR TITLE
feat: Runtime Params

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1187

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/helpers/GitInfoMaterialCard/GitInfoMaterial.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ import { ReceivedWebhookRedirectButton } from './ReceivedWebhookRedirectButton'
 
 const MissingPluginBlockState = importComponentFromFELibrary('MissingPluginBlockState', null, 'function')
 const RuntimeParamTabs = importComponentFromFELibrary('RuntimeParamTabs', null, 'function')
-const RuntimeParameters = importComponentFromFELibrary('RuntimeParameters', null, 'function')
+const RuntimeParametersV2 = importComponentFromFELibrary('RuntimeParametersV2', null, 'function')
```
Comment (eshankvaish) [inline src/components/common/helpers/GitInfoMaterialCard/GitInfoMaterial.tsx:55]: I think we will be deleting v1, right?

File: src/components/CIPipelineN/CreatePluginModal/utils.tsx [hunk 2]
```diff
@@ -156,6 +156,7 @@ const parseInputVariablesIntoCreatePluginPayload = (
         valueType: variable.variableType,
         referenceVariableName: variable.refVariableName,
         isExposed: true,
+        // TODO: handle file type here
```
Comment (eshankvaish) [inline src/components/CIPipelineN/CreatePluginModal/utils.tsx:159]: this one seems to be left

File: src/components/CIPipelineN/TaskDetailComponent.tsx [hunk 3]
```diff
@@ -273,7 +273,9 @@ export const TaskDetailComponent = () => {
                 </div>
                 <hr />
                 {selectedStep.stepType === PluginType.INLINE ? (
-                    <CustomInputOutputVariables type={PluginVariableType.INPUT} />
+                    <>
```
Comment (eshankvaish) [inline src/components/CIPipelineN/TaskDetailComponent.tsx:276]: do we need the fragments?

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/common/helpers/GitInfoMaterialCard/GitInfoMaterial.tsx (hunk 1 above)
Comment (RohitRaj011) [reply src/components/common/helpers/GitInfoMaterialCard/GitInfoMaterial.tsx:55]: yes

File: src/components/CIPipelineN/VariableDataTable/constants.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    DynamicDataTableHeaderType,
+    SelectPickerOptionType,
+    VariableTypeFormat,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { PluginVariableType } from '@Components/ciPipeline/types'
+
+import { VariableDataKeys } from './types'
+
+export const getVariableDataTableHeaders = (
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/components/CIPipelineN/VariableDataTable/constants.ts:11]: its a util but its returning a constant config

File: src/components/CIPipelineN/VariableDataTable/helpers.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Tooltip } from '@devtron-labs/devtron-fe-common-lib'
+
+import { ReactComponent as Var } from '@Icons/ic-var-initial.svg'
+import { TIPPY_VAR_MSG } from '../Constants'
+
+export const getSystemVariableIcon = () => (
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/components/CIPipelineN/VariableDataTable/helpers.tsx:6]: helpers are basically render utils -> helpers.tsx that contain JSX elements
utils are normal functions

**Review by eshankvaish — DISMISSED** [review]

File: src/components/ciPipeline/ciPipeline.service.ts [hunk 6]
```diff
@@ -23,13 +23,15 @@ import {
...
     PipelineBuildStageType,
+    VariableTypeFormat,
 } from '@devtron-labs/devtron-fe-common-lib'
 import { Routes, SourceTypeMap, TriggerType, ViewType } from '../../config'
 import { getSourceConfig, getWebhookDataMetaConfig } from '../../services/service'
 import { CiPipelineSourceTypeBaseOptions } from '../CIPipelineN/ciPipeline.utils'
 import { PatchAction } from './types'
 import { safeTrim } from '../../util/Util'
 import { ChangeCIPayloadType } from '../workflowEditor/types'
+import { MutableRefObject } from 'react'
```
Comment (vivek-devtron) [inline src/components/ciPipeline/ciPipeline.service.ts:34]: Please remove this unused import

File: src/components/CIPipelineN/VariableDataTable/utils.tsx [hunk 7]
```diff
@@ -0,0 +1,629 @@
...
+            variableDescription,
+            fileInfo,
+        } = customState
+
+        const variableDetail: VariableType = {
+            ...selectedIOVariable,
+            value: data.val.value,
+            format: data.format.value as VariableTypeFormat,
+            name: data.variable.value,
+            description: type === PluginVariableType.INPUT ? variableDescription : data.val.value,
```
Comment (vivek-devtron) [inline src/components/CIPipelineN/VariableDataTable/utils.tsx:523]: Why are we adding the value in description?

File: src/components/CIPipelineN/VariableDataTable/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,629 @@
...
+                choices.length
+            ) {
+                variableDetail.valueConstraint = {
+                    ...variableDetail.valueConstraint,
+                    choices: choices.map(({ value }) => value),
+                    blockCustomValue,
+                }
+            }
+
+            if (variableDetail.format === VariableTypeFormat.FILE && fileInfo) {
```
Comment (vivek-devtron) [inline src/components/CIPipelineN/VariableDataTable/utils.tsx:543]: We should change it to else if

File: src/components/CIPipelineN/VariableDataTable/VariableConfigOverlay.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+                    data-testid="ask-value-is-required"
+                >
+                    <Tooltip
+                        alwaysShowTippyOnHover
+                        className="w-200"
+                        placement="bottom-start"
+                        content={
+                            <div className="fs-12 lh-18">
+                                <p className="m-0 fw-6 cn-0">Value is required</p>
+                                <p className="m-0 cn-50">Get this tooltip from Utkarsh</p>
```
[the remark this answers is not in the document]
Comment (vivek-devtron) [reply src/components/CIPipelineN/VariableDataTable/VariableConfigOverlay.tsx:74]: Is it still pending?

[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/components/CIPipelineN/VariableDataTable/VariableConfigOverlay.tsx:74]: Done in https://github.com/devtron-labs/dashboard/pull/2289/

File: src/components/CIPipelineN/VariableDataTable/utils.tsx (hunk 7 above)
Comment (RohitRaj011) [reply src/components/CIPipelineN/VariableDataTable/utils.tsx:523]: in case of OutputVariables value column is being used as description

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**